### DISCUSSION TITLE
str_to_date 格式转换问题

### DISCUSSION THREAD

**Discussion by zzz222z** [apache/doris#11668] [A - General / Q&A]
select str_to_date('Tue Jul 12 23:23:23 2022', '%a %b %e %T %Y '); 这样得结果是null，是bug还是我得写法有问题呢

**Comment by hf200012** [comment]
select str_to_date('2014-12-21 12:34:56', '%Y-%m-%d %H:%i:%s');

**Reply by zzz222z** [reply]
%H:%i:%s 这个可以替换 %T，意思是%T不是这样用的吗（Doris官网：%T | 时间, 24-小时 (hh:mm:ss)）

**Comment by LemonLiTree** [comment]
你这边有两个问题：
1，str_to_date只是把类似于'2014-12-21 12:34:56'这样的时间字符串转化为date类型的时间戳。
2，如果你想要你那种格式的，可以先使用str_to_date转化为date类型，外面包一层date_format()函数来实现你要的那种格式。
另外，%T就是hh:mm:ss格式

**Reply by zzz222z** [reply]
1、select str_to_date('Tue Jul 12 2022 02:00:00', '%a %b %e %Y %H:%i:%s ');	这样写 是可以的呢。
我其实就是想把这样的Tue Jul 12 20:00:45 CST 2022 转换成2022-07-12 20:00:45 这样的，但是发现 时间只有用%H:%i:%s可以，用%T不行
